spring-pulsar/src/main/java/org/springframework/pulsar/core/DefaultSchemaResolver.java

```diff
@@ -130,7 +130,11 @@ public <T> Schema<T> getSchema(Class<?> messageClass, boolean returnDefault) {
 
 	@Nullable
 	private Schema<?> getCustomSchemaOrMaybeDefault(Class<?> messageClass, boolean returnDefault) {
-		return this.customSchemaMappings.getOrDefault(messageClass, (returnDefault ? Schema.BYTES : null));
+		Schema<?> schema = this.customSchemaMappings.get(messageClass);
+		if (schema == null && returnDefault) {
+			return messageClass != null ? Schema.JSON(messageClass) : Schema.BYTES;
```

Comment:
Wdyt about moving this defaulting picking logic into a protected method so that users could override just this piece of the resolver easily (custom resolver)? That would also be the place where we handle the "failed to do the Bar<T>..."

Reply:
Sure